### PR TITLE
Support config-server-encrypt-value

### PR DESCRIPTION
**Pull request by glyn**
This operation no longer requires a service key to be associated with the
config server. Rather than authenticating and authorising the encryption
request using an OAuth token with client credentials grant type, use an OAuth
token with resource owner password credentials grant type (obtained via an
equivalent of `cf oauth- token`).

* Factor out utilities for resolving a service instance URL from a service
  name and for parsing the output of an equivalent of `cf oauth-token` and
  returning just the OAuth token.
* Plug some holes in the tests.

[#143145493]

Refactoring

delete dead code

Test improvements

### CODE REVIEW

File: eureka/list_test.go [hunk 1]
```diff
@@ -56,94 +60,66 @@ var _ = Describe("Service Registry List", func() {
 		output, err = eureka.ListWithResolver(fakeCliConnection, "some-service-registry", fakeAuthClient, fakeResolver)
 	})
 
-	Context("when the service is not found", func() {
+	Context("but the access token is not available", func() {
```
Comment (georgeharley) [inline eureka/list_test.go]: Minor point: should this be "when the access token is not available"?

File: eureka/list_test.go [hunk 2]
```diff
@@ -56,94 +60,66 @@ var _ = Describe("Service Registry List", func() {
...
-			Expect(fakeCliConnection.GetServiceArgsForCall(0)).To(Equal("some-service-registry"))
+			Expect(fakeCliConnection.AccessTokenCallCount()).To(Equal(1))
 			Expect(err).To(HaveOccurred())
-			Expect(err).To(MatchError("Service registry instance not found: some error"))
+			Expect(err).To(MatchError("Access token not available: some access token error"))
 		})
 	})
 
-	Context("when the service is found", func() {
+	Context("and the access token is available", func() {
```
Comment (georgeharley) [inline eureka/list_test.go]: "when the access token is available"?

File: eureka/list_test.go (hunk 1 above)
Comment (glyn) [inline eureka/list_test.go]: Yes. Fixing.

File: eureka/list_test.go (hunk 2 above)
Comment (glyn) [inline eureka/list_test.go]: Yes. Fixing.

**Pull request merged**